fredsplit (#28)
SpecialeBajs/speciale-rapport: article/Method/alpha-k-layer-combination.tex

Based on the results from the node degree experiments seen in \autoref{subsec:degree-experiment}, we decided to make an algorithm that calculated each layers' effect based on how they performed compared to each other.
It could clearly be seen that some layers performed substantially better than others and therefore the worst layers might have a negative effect on the final embedding.
\\

> The algorithms looks at the performance of each individual layers and then based on how many percent's worse they perform compared to the best layer they will get a penalty in their effect and the other layers will get an increase.

Reviewer: ```suggestion
The algorithm looks at the performance of each individual layer and then based on how much worse they perform compared to the best layer they will get a penalty in their effect and the other layers will get an increase.
```

> Two algorithms have been created where one has a more aggressive approach, that can completely remove certain layers, and a more balanced approach, where each layer will be ensured influence.

Reviewer: ```suggestion
Two algorithms have been created where one has a more aggressive approach, that can completely remove certain layers, and the other one has a more balanced approach, where each layer will be ensured influence.
```

> An example of the aggressive algorithm can be seen on \autoref{fig:fredsplitAgg} and and the balanced algorithm on \autoref{fig:fredsplitBal}.

Reviewer: ```suggestion
An example of the aggressive algorithm can be seen on \autoref{fig:fredsplitAgg} and an example of the balanced algorithm can be seen on \autoref{fig:fredsplitBal}.
```

> Given four layers on \autoref{fig:fredsplitAgg} each layer will have 25\% effect on the final embedding at the start.

Reviewer: ```suggestion
Given four layers on \autoref{fig:fredsplitAgg} each layer will have 25\% effect on the final embedding initially.
```

The worst layer performs 15\% worse than the best layer, we therefore remove 15 from 25 to get its final effect of 10.

> Meanwhile, the 15 is split evenly among the other 3 layers who now have an effect of 30\%.

Reviewer: ```suggestion
Meanwhile, the 15 is split evenly among the 3 other layers who now have an effect of 30\%.
```

The same process is then repeated but the worst performing layer from the previous iteration is excluded.
This continues until all layers have gotten an effect penalty except the best performing layer.
This can result in one or more layers having 0 \% effect in the end.
In the balanced version the worst-performing layer is not excluded from the previous iteration.
\\
On \autoref{alg:aggresive-layereffect} the pseudocode for the aggressive algorithm can be seen.

> The algorithm is given a list of how much worse each layer performed in percentage compared to the best performing layer in the same list.

Reviewer: ```suggestion
The algorithm is given a list of how much worse each layer performed in percentages compared to the best performing layer in the same list.
```

This can result in one or more layers having 0 \% effect in the end.
A more balanced version can be done by not excluding the worst-performing layer from the previous iteration.

On \autoref{alg:layereffect} the pseudocode for the algorithm can be seen.
The algorithm is given a list of how much worse each layer performed in percentage compared to the best performing layer in the same list.
At the start each layers' influence \textit{Inf} is set to be the same.
After this the list \textit{L} is iterated over and the last element gets popped.
Since L is ordered the last element is the worst performing one.
Based on the start influence and how bad the layer has performed its new lower influence is calculated as well as the other layers influence increased.
On \autoref{fig:fredsplitAgg} and \autoref{fig:fredsplitBal} examples of the algorithm can be seen.

\begin{algorithm}

>     \caption{Algorithm for layer effect based on performance}

Reviewer 2: ```suggestion
    \caption{Algorithm for the aggressive layer effect based on performance}
```
